src/Microsoft.Kusto.ServiceLayer/QueryExecution/Batch.cs

```diff
@@ -266,6 +268,12 @@ public async Task Execute(ReliableDataSourceConnection conn, CancellationToken c
             {
                 await DoExecute(conn, cancellationToken);
             }
+            catch (KustoUnauthorizedException)
```

Comment:
Anyway to make this non-Kusto specific. Can the processing of the exception be passed to the DataSource?

Why a retry once. Can we make this more generic such that it can have multiple retries if needed.